### PR TITLE
[ENH] OWSetEnrichment: use filter proxy model

### PR DESCRIPTION
**Pull request by JakaKokosar**
- Re-enable string search and move filtering into a filter proxy model. 
- Remove UserWarning messages by using Error and Warning class.

### CODE REVIEW

**Comment by JakaKokosar** [comment]
@markotoplak

File: orangecontrib/bio/widgets3/OWSetEnrichment.py [hunk 1]
```diff
@@ -24,11 +24,12 @@
 )
 from AnyQt.QtCore import (
     Qt, QRect, QSize, QModelIndex, QStringListModel, QThread, QThreadPool,
-    Slot
+    Slot, QSortFilterProxyModel
 )
 
 import Orange
 from Orange.widgets.utils.datacaching import data_hints
+from Orange.widgets.utils.messages import UnboundMsg
```
Comment (ales-erjavec) [inline orangecontrib/bio/widgets3/OWSetEnrichment.py:32]: You can use `from Orange.widgets import Msg` and remove the reassign below.

File: orangecontrib/bio/widgets3/OWSetEnrichment.py [hunk 2]
```diff
@@ -825,16 +908,14 @@ def item(value=None, tooltip=None, user=None):
...
-
         self.progressBarFinished()
         self.setStatusMessage("")
 
     def _updatesummary(self):
         state = self.state
         if state is None:
-            self.error(0,)
-            self.warning(0)
+            self.Error.no_data_onInput()
```
Comment (ales-erjavec) [inline orangecontrib/bio/widgets3/OWSetEnrichment.py:917]: Don't show an error here.

File: orangecontrib/bio/widgets3/OWSetEnrichment.py [hunk 3]
```diff
@@ -752,11 +832,8 @@ def item(value=None, tooltip=None, user=None):
                 si.setData(value, Qt.UserRole)
             return si
 
-        model = QStandardItemModel()
-        model.setSortRole(Qt.UserRole)
-        model.setHorizontalHeaderLabels(
-            ["Category", "Term", "Count", "Reference count", "p-value",
-             "FDR", "Enrichment"])
+        model = self.source_model
```
Comment (ales-erjavec) [inline orangecontrib/bio/widgets3/OWSetEnrichment.py:835]: This model is only appended to. Running multiple enrichement tasks one after the other results in multiplied rows in the model and view:
screen shot 2018-01-11 at 14 20 15

**Review by ales-erjavec — CHANGES_REQUESTED** [review]

**Comment by ales-erjavec** [comment]
Closed in favor of gh-122

**Pull request closed without merging**